extras/rya.export/export.mongo/src/main/java/org/apache/rya/export/mongo/MongoRyaStatementStore.java

```diff
@@ -0,0 +1,142 @@
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.rya.api.domain.RyaStatement;
+import org.apache.rya.api.persist.RyaDAOException;
+import org.apache.rya.export.api.metadata.MergeParentMetadata;
+import org.apache.rya.export.api.metadata.ParentMetadataExistsException;
+import org.apache.rya.export.api.store.AddStatementException;
+import org.apache.rya.export.api.store.ContainsStatementException;
+import org.apache.rya.export.api.store.RemoveStatementException;
+import org.apache.rya.export.api.store.RyaStatementStore;
+import org.apache.rya.export.api.store.UpdateStatementException;
+import org.apache.rya.export.mongo.parent.MongoParentMetadataRepository;
+import org.apache.rya.mongodb.MongoDBRyaDAO;
+import org.apache.rya.mongodb.dao.SimpleMongoDBStorageStrategy;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.Cursor;
+import com.mongodb.DB;
+import com.mongodb.DBObject;
+import com.mongodb.MongoClient;
+
+/**
+ * Mongo implementation of {@link RyaStatementStore}.  Allows for exporting and
+ * importing rya statements from MongoDB.
+ */
+public class MongoRyaStatementStore implements RyaStatementStore {
+    public static final String TRIPLES_COLLECTION = "rya__triples";
+    public static final String METADATA_COLLECTION = "parent_metadata";
+    protected final SimpleMongoDBStorageStrategy adapter;
+    protected final DB db;
+
+    private final String ryaInstanceName;
+    private final MongoClient client;
+    private final MongoDBRyaDAO dao;
+    private final MongoParentMetadataRepository parentMetadataRepo;
+
+    /**
+     * Creates a new {@link MongoRyaStatementStore}.
+     * @param client - The client to connect to Mongo. (not null)
+     * @param ryaInstance - The rya instance to connect to. (not null)
+     * @param dao - The {@link MongoDBRyaDAO} to use to access statements. (not null)
+     */
+    public MongoRyaStatementStore(final MongoClient client, final String ryaInstance, final MongoDBRyaDAO dao) {
+        this.client = checkNotNull(client);
+        ryaInstanceName = checkNotNull(ryaInstance);
+        this.dao = checkNotNull(dao);
+        db = this.client.getDB(ryaInstanceName);
+        adapter = new SimpleMongoDBStorageStrategy();
+        parentMetadataRepo = new MongoParentMetadataRepository(client, ryaInstance);
+    }
+
+    @Override
+    public Iterator<RyaStatement> fetchStatements() {
+        final Cursor cur = db.getCollection(TRIPLES_COLLECTION).find().sort(new BasicDBObject(TIMESTAMP, 1));
+        final List<RyaStatement> statements = new ArrayList<>();
+        while(cur.hasNext()) {
+            final RyaStatement statement = adapter.deserializeDBObject(cur.next());
+            statements.add(statement);
+        }
+        return statements.iterator();
+    }
+
+    @Override
+    public void addStatement(final RyaStatement statement) throws AddStatementException {
+        try {
+            dao.add(statement);
```

Comment:
accumulo marks things for deletion instead of flat out deleting them.  Which means if you go to add a statement back into accumulo before a compaction, it won't get added back.  This is not a problem in mongo.